src/hazelweb/gui/UHCode.re

```diff
@@ -250,23 +250,24 @@ let key_handlers =
 };
 
 let box_table: WeakMap.t(UHBox.t, list(Vdom.Node.t)) = WeakMap.mk();
-let rec view_of_box = (box: UHBox.t): list(Vdom.Node.t) => {
+let rec view_of_box = (program: Program.t, box: UHBox.t): list(Vdom.Node.t) => {
```

Comment:
Couple issues here:
- In general you should try to minimize what you pass into only what's necessary. In this case, that's just the assert map.
- First issue aside, this doesn't work with the memoization. The memoization table is only indexed on the `UHBox.t`, so if the assert map changes, there will still be a memo cache hit and the assert map change won't be reflected. Rather than embedding the assert decorations in the code text, we should be rendering them as layers on top of the text like we're doing with other decorations. Talk to me about this.